### ISSUE TITLE
Typedoc options documentation unclear about typedocs.json usage and required parameters

### ISSUE THREAD

**Issue by JHKennedy4** [TypeStrong/typedoc#1343]
Trying out Typedocs for the first time today, I was able to construct a working command using the CLI. When I tried to convert that command to a typedoc.json configuration, I would see the help message print every time I ran the command.

What I was doing:
typedoc.json
```
{
  "inputFiles": ["./src"],
  "mode": "modules",
  "out": "doc"
}
```

then
```
$ typedoc --options typedoc.json
```
Which output the following. I assumed that the --options configuration could be used to replace any parameter, so it wasn't obvious to me that I was using it wrong, but there were no other error messages.
```
Usage:
 typedoc --mode modules --out path/to/documentation path/to/sourcefiles
```

What worked:
typedoc.json
```
{
  "mode": "modules",
  "out": "doc"
}
```
then
```
$ typedoc --options typedoc.json ./src
```

I'm still not clear on the difference between the required source directory and inputFiles. I expected if I had one, I wouldn't need the other.

Not sure if I missed something obvious in the documentation, but I think instructions at the top of the docs clearly explaining the required parameters and a usage example in the Options documentation would go a long way.

If nothing needs changing, I figure this issue might at least help the next person who's trying to figure out why their typedoc.json won't load.

**Comment by JHKennedy4** [comment]
Re-reading the inputFiles doc, I think this could be a bug where typedoc is not looking for inputFiles in the options configuration before checking whether they have been provided via the command line.

**Comment by Gerrit0** [comment]
I believe this is due to #1265, and is now fixed in 0.18.0. There was a bug with inputFiles always being overwritten by what tsconfig set.
